src/dev/flang/be/interpreter/Intrinsics.java

```diff
@@ -155,6 +171,31 @@ public static Callable call(Interpreter interpreter, Clazz innerClazz)
     return result;
   }
 
+  /**
+   * Checks the file descriptors stack and expands it as necessary.
+   *
+   * @return the next available file descriptor.
+   */
+  private static long getAvailableFileDescriptor()
+  {
+    if (_availableFileDescriptors_.empty())
+      {
+        for(long i=10000;i<99999;i++)
+          {
+            _availableFileDescriptors_.push(i);
+          }
+      }
+    if (_availableFileDescriptors_.size()==1)
+      {
+        long lastFD = _availableFileDescriptors_.pop();
+        for(long i=lastFD+1;i<lastFD+89999;i++)
+          {
+            _availableFileDescriptors_.push(i);
+          }
+        _availableFileDescriptors_.push(lastFD);
+      }
+    return _availableFileDescriptors_.pop();
```

Comment:
This code is IMHO too complex, and probably even wrong: The first loop pushes 10000, 10001, 10002, ... 99998. After 89998 have been popped, the stack contains only 100000, then the second loop pushes 10001, 10002, ... 99998, so we get all of the same file descriptors again. 

My suggestions: 

- add a field _maxFileDescriptor_
- on getAvailableFileDescriptor, check for empty() and if it is empty, push only 1 new fd with the value of _maxFileDescriptor_ and increment _maxFileDescriptor_
- make getAvailableFileDescriptor synchronized such that it will work with several threads
- in case you need any constants other than 0, 1, -1, declare them as static final fields and explain what they are (why does the first file descriptor start with 10000, why not 0? 
- maybe change the name of this method to just `allocFileDescriptor`
- add a method `releaseFileDescriptor` instead of doing the `push` inline.

Reply:
Usually 0, 1, 2 are reserved for standard in, out and err that is why I chose to start at 10000 (could be any number for that matter).